airbyte-config/config-models/src/test/java/io/airbyte/config/helpers/StateMessageHelperTest.java

```diff
@@ -13,31 +13,52 @@
 import io.airbyte.protocol.models.AirbyteStateMessage.AirbyteStateType;
 import io.airbyte.protocol.models.AirbyteStreamState;
 import io.airbyte.protocol.models.StreamDescriptor;
+import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 import org.assertj.core.api.Assertions;
 import org.junit.jupiter.api.Test;
 
 public class StateMessageHelperTest {
 
+  private static final boolean USE_STREAM_CAPABLE_STATE = true;
+  private static final boolean DONT_USE_STREAM_CAPABALE_STATE = false;
+
   @Test
   public void testEmpty() {
-    final Optional<StateWrapper> stateWrapper = StateMessageHelper.getTypedState(null);
+    final Optional<StateWrapper> stateWrapper = StateMessageHelper.getTypedState(null, USE_STREAM_CAPABLE_STATE);
+    Assertions.assertThat(stateWrapper).isEmpty();
+  }
+
+  @Test
+  public void testEmptyList() {
+    final Optional<StateWrapper> stateWrapper = StateMessageHelper.getTypedState(Jsons.arrayNode(), USE_STREAM_CAPABLE_STATE);
     Assertions.assertThat(stateWrapper).isEmpty();
   }
 
   @Test
   public void testLegacy() {
-    final Optional<StateWrapper> stateWrapper = StateMessageHelper.getTypedState(Jsons.emptyObject());
+    final Optional<StateWrapper> stateWrapper = StateMessageHelper.getTypedState(Jsons.emptyObject(), USE_STREAM_CAPABLE_STATE);
     Assertions.assertThat(stateWrapper).isNotEmpty();
     Assertions.assertThat(stateWrapper.get().getStateType()).isEqualTo(StateType.LEGACY);
   }
 
   @Test
   public void testLegacyInList() {
     final Optional<StateWrapper> stateWrapper = StateMessageHelper.getTypedState(Jsons.jsonNode(
-        Lists.newArrayList(
-            Map.of("Any", "value"))));
+        List.of(
+            Map.of("Any", "value"))),
+        USE_STREAM_CAPABLE_STATE);
+    Assertions.assertThat(stateWrapper).isNotEmpty();
+    Assertions.assertThat(stateWrapper.get().getStateType()).isEqualTo(StateType.LEGACY);
```

Comment:
I think this test should also verify that `stateWrapper.get().getLegacyState()` is set to the correct json blob

Reply:
Done